src/api/tee_host.rs

```diff
@@ -321,3 +321,69 @@ pub unsafe fn add_shared_pages(
     ecall_send(&msg)?;
     Ok(())
 }
+
+/// Invalidates the pages in the specified range of guest physical address space.
+pub fn block_pages(vmid: u64, guest_addr: u64, len: u64) -> Result<()> {
+    let msg = SbiMessage::TeeHost(TvmBlockPages {
+        guest_id: vmid,
+        guest_addr,
+        len,
+    });
+    // Safety: Each page is marked present in the TVM's page table and it's uniquely owned
+    // by the TVM or the host, depending on the page state.
```

Comment:
For safety comments, the point is to convey why the `unsafe` code is safe w.r.t. the Rust memory model / undefined behavior. In this case, all these ECALLs are "safe" because we're manipulating a guest's address space, not our own.